CMakeLists.txt

```diff
@@ -74,5 +74,14 @@ IF(Trilinos_ENABLE_DataTransferKitMoabAdapters)
 ENDIF()
 
 ##---------------------------------------------------------------------------##
+## Check that there are no trailing whitespace
+##---------------------------------------------------------------------------##
+
+ADD_TEST(
+    NAME check_no_trailing_whitespace
+    COMMAND sh -c "grep --exclude=*.git* -rl '[[:blank:]]$' ${CMAKE_SOURCE_DIR} test $? -eq 1"
+)
```

Comment:
Here is one way of doing that.  It will fail right now because you only edited the *.{cpp,cpp} files in your commit.